code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -63,6 +63,9 @@
 	if(!reagents.has_any_reagents(HYPERZINES))
 		if(!shoes)
 			. *= NO_SHOES_SLOWDOWN
+		if(addicted_chems.has_any_reagents(HYPERZINES))
+			var/datum/reagent/R = addicted_chems.get_reagent_by_type(/datum/reagent/hyperzine)
+			. *= min(2,R.volume/5)
```

Comment:
Be honest, how much did you test this one?

Reply:
Not yet, will update this comment when I do. PR is in draft for a reason.

Reply:
Just did now, doing 100u of hyperzine consecutively (as opposed to having it all in you at once) will cause you to move at about the same speed as a diona, which this is capped at.